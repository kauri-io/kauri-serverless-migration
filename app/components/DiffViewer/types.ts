export interface IDiff {
    count: number
    value: string
    added: boolean
    removed: boolean
    compareValue: string
}
