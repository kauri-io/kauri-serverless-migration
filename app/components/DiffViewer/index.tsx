import styled from 'styled-components'
import ReactDiffViewer from 'react-diff-viewer'

interface IProps {
    current: string
    proposed: string
}

const Content = styled.div`
    font-size: ${props => props.theme.fontSizes[3]}px;
    width: 100%;
    background: white;
    word-break: break-all;
`

const Line = styled.span`
    font-family: 'Roboto';
`

const styleObj = {
    diffAdded: {
        maxWidth: '95vh',
        padding: 4,
    }, // style object
    diffRemoved: {
        maxWidth: '95vh',
        padding: 4,
    }, // style object
    gutter: {
        display: 'none',
    },
    line: {
        fontFamily: 'Roboto',
    },
    lineNumber: {
        display: 'none',
    }, // style object
    marker: {
        display: 'none',
    }, // style object
    wordAdded: { fontWeight: 700, color: '#006d27 !important' }, // style object
    wordRemoved: { fontWeight: 700, color: '#960016 !important' }, // style object
}

const renderLine = (str: string) => {
    return <Line>{str}</Line>
}

const Diffs = ({ current, proposed }: IProps) => {
    return (
        <Content className="DiffViewer">
            <ReactDiffViewer
                oldValue={current}
                newValue={proposed}
                splitView={false}
                styles={styleObj}
                renderContent={renderLine}
            />
        </Content>
    )
}

export default Diffs
