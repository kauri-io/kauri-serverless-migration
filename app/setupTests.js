import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
