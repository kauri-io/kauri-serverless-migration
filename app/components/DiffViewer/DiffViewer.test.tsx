import DiffViewer from './index'
import { mountWithTheme } from '../../setupTests'

let props = {
    current: '# this is a test',
    proposed: '# this is an update to the test',
}

describe('components/DiffViewer', () => {
    it('should match snapshot', () => {
        const wrapper = mountWithTheme(<DiffViewer {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
