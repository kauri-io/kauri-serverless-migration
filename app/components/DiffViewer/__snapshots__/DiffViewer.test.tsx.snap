// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/DiffViewer should match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "DEFAULT_CARD_WIDTH": 288,
      "MAIN_BODY_WIDTH": 1242,
      "bg": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "bgPrimary": "#1E2428",
      "bgSecondary": "#1E3D3B",
      "borderTextArea": "#979797",
      "breakpoints": Array [
        "500px",
        "52em",
        "1242px",
      ],
      "colors": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
      "disabledBackgroundColor": "#D6D6D6",
      "disabledTextColor": "#9B9B9B",
      "divider": "#EBEBEB",
      "errorRedColor": "#C03030",
      "fontSizes": Array [
        11,
        13,
        14,
        16,
        18,
        20,
        22,
        28,
        32,
      ],
      "fontWeight": Array [
        300,
        "normal",
        500,
        700,
      ],
      "hoverTextColor": "#108E72",
      "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
      "padding": "calc((100vw - 1242px) / 2)",
      "paddingTop": "2.5em",
      "primary": "#0BA986",
      "primaryColor": "#0BA986",
      "primaryDark": "#11856B",
      "primaryTextColor": "#1E2428",
      "secondary": "#11856B",
      "secondaryBlue": "rgb(81, 123, 162, 1)",
      "secondaryBlueDark": "rgb(81, 118, 142, 1)",
      "secondaryColor": "#1E3D3B",
      "secondaryTextColor": "#283035",
      "space": Array [
        0,
        10,
        15,
        30,
        60,
      ],
      "tertiaryBackgroundColor": "#F7F7F7",
      "textPrimary": "#1E2428",
      "white": "#FFFFFF",
    }
  }
>
  <Diffs
    current="# this is a test"
    proposed="# this is an update to the test"
  >
    <DiffViewer__Content
      className="DiffViewer"
    >
      <StyledComponent
        className="DiffViewer"
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "DiffViewer__Content-sc-1yaf9sg-0",
              "isStatic": false,
              "lastClassName": "ecffEv",
              "rules": Array [
                "font-size:",
                [Function],
                "px;width:100%;background:white;word-break:break-all;",
              ],
            },
            "displayName": "DiffViewer__Content",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "DiffViewer__Content-sc-1yaf9sg-0",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="DiffViewer DiffViewer__Content-sc-1yaf9sg-0 ecffEv"
        >
          <DiffViewer
            disableWordDiff={false}
            hideLineNumbers={false}
            highlightLines={Array []}
            newValue="# this is an update to the test"
            oldValue="# this is a test"
            renderContent={[Function]}
            splitView={false}
            styles={
              Object {
                "diffAdded": Object {
                  "maxWidth": "95vh",
                  "padding": 4,
                },
                "diffRemoved": Object {
                  "maxWidth": "95vh",
                  "padding": 4,
                },
                "gutter": Object {
                  "display": "none",
                },
                "line": Object {
                  "fontFamily": "Roboto",
                },
                "lineNumber": Object {
                  "display": "none",
                },
                "marker": Object {
                  "display": "none",
                },
                "wordAdded": Object {
                  "color": "#006d27 !important",
                  "fontWeight": 700,
                },
                "wordRemoved": Object {
                  "color": "#960016 !important",
                  "fontWeight": 700,
                },
              }
            }
          >
            <table
              className="css-wu4c77-diff-container"
            >
              <tbody>
                <Component
                  content={
                    Array [
                      <span
                        className="css-15tgi4e-word-diff"
                      >
                        <ForwardRef(DiffViewer__Line)>
                          # this is 
                        </ForwardRef(DiffViewer__Line)>
                      </span>,
                      <span
                        className="css-15tgi4e-word-diff css-1uhvp69-word-removed"
                      >
                        <ForwardRef(DiffViewer__Line)>
                          a
                        </ForwardRef(DiffViewer__Line)>
                      </span>,
                      undefined,
                      <span
                        className="css-15tgi4e-word-diff"
                      >
                        <ForwardRef(DiffViewer__Line)>
                           
                        </ForwardRef(DiffViewer__Line)>
                      </span>,
                      undefined,
                      <span
                        className="css-15tgi4e-word-diff"
                      >
                        <ForwardRef(DiffViewer__Line)>
                          test
                        </ForwardRef(DiffViewer__Line)>
                      </span>,
                    ]
                  }
                  hideLineNumbers={false}
                  hightlightLines={Array []}
                  key="0"
                  leftLineNumber={1}
                  removed={true}
                  renderContent={[Function]}
                  rightLineNumber={true}
                  styles={
                    Object {
                      "diffAdded": "css-1smcabo-diff-added",
                      "diffContainer": "css-wu4c77-diff-container",
                      "diffRemoved": "css-oy5qt9-diff-removed",
                      "gutter": "css-1uwdc4n-gutter",
                      "hightlightedGutter": "css-lzf6ep-hightlighted-gutter",
                      "hightlightedLine": "css-h1kdcl-hightlighted-line",
                      "leftGutter": "css-1rowcki-left-gutter",
                      "line": "css-7gnhnh-line",
                      "marker": "css-hvdmkm-marker",
                      "rightGutter": "css-84qt4m-right-gutter",
                      "wordAdded": "css-1e0k4cm-word-added",
                      "wordDiff": "css-15tgi4e-word-diff",
                      "wordRemoved": "css-1uhvp69-word-removed",
                    }
                  }
                >
                  <tr
                    className="css-7gnhnh-line"
                  >
                    <td
                      className="css-1uwdc4n-gutter css-1rowcki-left-gutter css-oy5qt9-diff-removed"
                      onClick={[Function]}
                    >
                      <LineNumber
                        lineNumber={1}
                        prefix="L"
                      >
                        <pre
                          id="L-1"
                        >
                          1
                        </pre>
                      </LineNumber>
                    </td>
                    <td
                      className="css-1uwdc4n-gutter css-84qt4m-right-gutter css-oy5qt9-diff-removed"
                      onClick={[Function]}
                    />
                    <td
                      className="css-hvdmkm-marker css-oy5qt9-diff-removed"
                    >
                      <pre>
                        -
                      </pre>
                    </td>
                    <td
                      className="css-oy5qt9-diff-removed"
                    >
                      <pre>
                        <span
                          className="css-15tgi4e-word-diff"
                          key="0"
                        >
                          <DiffViewer__Line>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                    "isStatic": true,
                                    "lastClassName": "kynGKV",
                                    "rules": Array [
                                      "font-family:'Roboto';",
                                    ],
                                  },
                                  "displayName": "DiffViewer__Line",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                  "target": "span",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <span
                                className="DiffViewer__Line-sc-1yaf9sg-1 kynGKV"
                              >
                                # this is 
                              </span>
                            </StyledComponent>
                          </DiffViewer__Line>
                        </span>
                        <span
                          className="css-15tgi4e-word-diff css-1uhvp69-word-removed"
                          key="1"
                        >
                          <DiffViewer__Line>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                    "isStatic": true,
                                    "lastClassName": "kynGKV",
                                    "rules": Array [
                                      "font-family:'Roboto';",
                                    ],
                                  },
                                  "displayName": "DiffViewer__Line",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                  "target": "span",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <span
                                className="DiffViewer__Line-sc-1yaf9sg-1 kynGKV"
                              >
                                a
                              </span>
                            </StyledComponent>
                          </DiffViewer__Line>
                        </span>
                        <span
                          className="css-15tgi4e-word-diff"
                          key="3"
                        >
                          <DiffViewer__Line>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                    "isStatic": true,
                                    "lastClassName": "kynGKV",
                                    "rules": Array [
                                      "font-family:'Roboto';",
                                    ],
                                  },
                                  "displayName": "DiffViewer__Line",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                  "target": "span",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <span
                                className="DiffViewer__Line-sc-1yaf9sg-1 kynGKV"
                              >
                                 
                              </span>
                            </StyledComponent>
                          </DiffViewer__Line>
                        </span>
                        <span
                          className="css-15tgi4e-word-diff"
                          key="5"
                        >
                          <DiffViewer__Line>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                    "isStatic": true,
                                    "lastClassName": "kynGKV",
                                    "rules": Array [
                                      "font-family:'Roboto';",
                                    ],
                                  },
                                  "displayName": "DiffViewer__Line",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                  "target": "span",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <span
                                className="DiffViewer__Line-sc-1yaf9sg-1 kynGKV"
                              >
                                test
                              </span>
                            </StyledComponent>
                          </DiffViewer__Line>
                        </span>
                      </pre>
                    </td>
                  </tr>
                </Component>
                <Component
                  added={true}
                  content={
                    Array [
                      <span
                        className="css-15tgi4e-word-diff"
                      >
                        <ForwardRef(DiffViewer__Line)>
                          # this is 
                        </ForwardRef(DiffViewer__Line)>
                      </span>,
                      undefined,
                      <span
                        className="css-15tgi4e-word-diff css-1e0k4cm-word-added"
                      >
                        <ForwardRef(DiffViewer__Line)>
                          an
                        </ForwardRef(DiffViewer__Line)>
                      </span>,
                      <span
                        className="css-15tgi4e-word-diff"
                      >
                        <ForwardRef(DiffViewer__Line)>
                           
                        </ForwardRef(DiffViewer__Line)>
                      </span>,
                      <span
                        className="css-15tgi4e-word-diff css-1e0k4cm-word-added"
                      >
                        <ForwardRef(DiffViewer__Line)>
                          update to the 
                        </ForwardRef(DiffViewer__Line)>
                      </span>,
                      <span
                        className="css-15tgi4e-word-diff"
                      >
                        <ForwardRef(DiffViewer__Line)>
                          test
                        </ForwardRef(DiffViewer__Line)>
                      </span>,
                    ]
                  }
                  hideLineNumbers={false}
                  hightlightLines={Array []}
                  key="0"
                  leftLineNumber={true}
                  renderContent={[Function]}
                  rightLineNumber={1}
                  styles={
                    Object {
                      "diffAdded": "css-1smcabo-diff-added",
                      "diffContainer": "css-wu4c77-diff-container",
                      "diffRemoved": "css-oy5qt9-diff-removed",
                      "gutter": "css-1uwdc4n-gutter",
                      "hightlightedGutter": "css-lzf6ep-hightlighted-gutter",
                      "hightlightedLine": "css-h1kdcl-hightlighted-line",
                      "leftGutter": "css-1rowcki-left-gutter",
                      "line": "css-7gnhnh-line",
                      "marker": "css-hvdmkm-marker",
                      "rightGutter": "css-84qt4m-right-gutter",
                      "wordAdded": "css-1e0k4cm-word-added",
                      "wordDiff": "css-15tgi4e-word-diff",
                      "wordRemoved": "css-1uhvp69-word-removed",
                    }
                  }
                >
                  <tr
                    className="css-7gnhnh-line"
                  >
                    <td
                      className="css-1uwdc4n-gutter css-1rowcki-left-gutter css-1smcabo-diff-added"
                      onClick={[Function]}
                    />
                    <td
                      className="css-1uwdc4n-gutter css-84qt4m-right-gutter css-1smcabo-diff-added"
                      onClick={[Function]}
                    >
                      <LineNumber
                        lineNumber={1}
                        prefix="R"
                      >
                        <pre
                          id="R-1"
                        >
                          1
                        </pre>
                      </LineNumber>
                    </td>
                    <td
                      className="css-hvdmkm-marker css-1smcabo-diff-added"
                    >
                      <pre>
                        +
                      </pre>
                    </td>
                    <td
                      className="css-1smcabo-diff-added"
                    >
                      <pre>
                        <span
                          className="css-15tgi4e-word-diff"
                          key="0"
                        >
                          <DiffViewer__Line>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                    "isStatic": true,
                                    "lastClassName": "kynGKV",
                                    "rules": Array [
                                      "font-family:'Roboto';",
                                    ],
                                  },
                                  "displayName": "DiffViewer__Line",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                  "target": "span",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <span
                                className="DiffViewer__Line-sc-1yaf9sg-1 kynGKV"
                              >
                                # this is 
                              </span>
                            </StyledComponent>
                          </DiffViewer__Line>
                        </span>
                        <span
                          className="css-15tgi4e-word-diff css-1e0k4cm-word-added"
                          key="2"
                        >
                          <DiffViewer__Line>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                    "isStatic": true,
                                    "lastClassName": "kynGKV",
                                    "rules": Array [
                                      "font-family:'Roboto';",
                                    ],
                                  },
                                  "displayName": "DiffViewer__Line",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                  "target": "span",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <span
                                className="DiffViewer__Line-sc-1yaf9sg-1 kynGKV"
                              >
                                an
                              </span>
                            </StyledComponent>
                          </DiffViewer__Line>
                        </span>
                        <span
                          className="css-15tgi4e-word-diff"
                          key="3"
                        >
                          <DiffViewer__Line>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                    "isStatic": true,
                                    "lastClassName": "kynGKV",
                                    "rules": Array [
                                      "font-family:'Roboto';",
                                    ],
                                  },
                                  "displayName": "DiffViewer__Line",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                  "target": "span",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <span
                                className="DiffViewer__Line-sc-1yaf9sg-1 kynGKV"
                              >
                                 
                              </span>
                            </StyledComponent>
                          </DiffViewer__Line>
                        </span>
                        <span
                          className="css-15tgi4e-word-diff css-1e0k4cm-word-added"
                          key="4"
                        >
                          <DiffViewer__Line>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                    "isStatic": true,
                                    "lastClassName": "kynGKV",
                                    "rules": Array [
                                      "font-family:'Roboto';",
                                    ],
                                  },
                                  "displayName": "DiffViewer__Line",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                  "target": "span",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <span
                                className="DiffViewer__Line-sc-1yaf9sg-1 kynGKV"
                              >
                                update to the 
                              </span>
                            </StyledComponent>
                          </DiffViewer__Line>
                        </span>
                        <span
                          className="css-15tgi4e-word-diff"
                          key="5"
                        >
                          <DiffViewer__Line>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                    "isStatic": true,
                                    "lastClassName": "kynGKV",
                                    "rules": Array [
                                      "font-family:'Roboto';",
                                    ],
                                  },
                                  "displayName": "DiffViewer__Line",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "DiffViewer__Line-sc-1yaf9sg-1",
                                  "target": "span",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <span
                                className="DiffViewer__Line-sc-1yaf9sg-1 kynGKV"
                              >
                                test
                              </span>
                            </StyledComponent>
                          </DiffViewer__Line>
                        </span>
                      </pre>
                    </td>
                  </tr>
                </Component>
              </tbody>
            </table>
          </DiffViewer>
        </div>
      </StyledComponent>
    </DiffViewer__Content>
  </Diffs>
</ThemeProvider>
`;
